
Student: Ismaa-eel Hendricks

In [ ]:
#load the dataset
import pandas as pd
import numpy as np


In [ ]:
#mount drive to colab
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#Data Selection

I have selected the data for:

[Car Insurance](https://www.kaggle.com/datasets/sagnik1511/car-insurance-data)

[Stroke Prediction Data Set](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset)










#First Choice

In [ ]:
#create file path
fpath = '/content/drive/MyDrive/coding dojo week 6/coding dojo projec2 /Car_Insurance_Claim.csv'
df = pd.read_csv(fpath)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   10000 non-null  int64  
 1   AGE                  10000 non-null  object 
 2   GENDER               10000 non-null  object 
 3   RACE                 10000 non-null  object 
 4   DRIVING_EXPERIENCE   10000 non-null  object 
 5   EDUCATION            10000 non-null  object 
 6   INCOME               10000 non-null  object 
 7   CREDIT_SCORE         9018 non-null   float64
 8   VEHICLE_OWNERSHIP    10000 non-null  int64  
 9   VEHICLE_YEAR         10000 non-null  object 
 10  MARRIED              10000 non-null  int64  
 11  CHILDREN             10000 non-null  int64  
 12  POSTAL_CODE          10000 non-null  int64  
 13  ANNUAL_MILEAGE       9043 non-null   float64
 14  VEHICLE_TYPE         10000 non-null  object 
 15  SPEEDING_VIOLATIONS  10000 non-null  

In [ ]:
#Description of the dataset

#how many cases are included in the dataset
length = len(df)
#how many features are in the dataset
features = df.shape[1]-1

print ("There are "+ str(len(df))+" cases in this dataset")
print ("There are {}".format(features)+" features in this dataset")


There are 10000 cases in this dataset
There are 18 features in this dataset


In [ ]:
df['OUTCOME'].value_counts()

0    6867
1    3133
Name: OUTCOME, dtype: int64

##First choice: [Car Insurance](https://www.kaggle.com/datasets/sagnik1511/car-insurance-data)

1. **Source of data**\
[www.kaggle.com](https://www.kaggle.com/)
2. **Brief description of data**\
Context:\
The company has shared its annual car insurance data. Now, you have to find out the real customer behaviors over the data.

3. **What is the target?**\
The outcome column indicates 1 if a customer has claimed his/her loan else 0.

4. **What does one row represent?** (A person? A business? An event? A product?)\
An Event
5. **Is this a classification or regression problem?**\
This is a classification problem

6. **How many features does the data have?**\
There are 18 features in this dataset

7. **How many rows are in the dataset?**\
There are 10 000 cases in this dataset

8. **What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset**?
\
One challenge could be dealing with data imbalance, which might necessitate the identification and application of appropriate techniques and hyperparameters to effectively model the imbalanced dataset. Additionally, determining the optimal approaches for data standardization, handling missing values through imputation, and selecting appropriate modeling strategies will be crucial for achieving the highest level of prediction accuracy

#Second choice

In [ ]:
#create file path
fpath = '/content/drive/MyDrive/coding dojo week 6/coding dojo projec2 /healthcare-dataset-stroke-data.csv'
dhc = pd.read_csv(fpath)
dhc.info(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In [ ]:
#how many cases are included in the dataset
length = len(dhc)
#how many features are in the dataset
features = dhc.shape[1]-1

print ("There are "+ str(len(dhc))+" cases in this dataset")
print ("There are {}".format(features)+" features in this dataset")

There are 5110 cases in this dataset
There are 11 features in this dataset


In [ ]:
dhc['stroke'].value_counts()

0    4861
1     249
Name: stroke, dtype: int64

##Second choice: [Stroke Prediction Data Set](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset)

1. **Source of data**\
[www.kaggle.com](https://www.kaggle.com/)
2. **Brief description of data**\
Context:\
According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.
This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status.

3. **What is the target?**\
Determine if the patient will have stroke or not. If yes, then 1, else 0. The target is the variable that a patientw will have a stroke or not.

4. **What does one row represent?** (A person? A business? An event? A product?)\
Each row in the data provides relavant information about the patient.
5. **Is this a classification or regression problem?**\
This is a classification problem.

6. **How many features does the data have?**\
There are 11 features in this dataset

7. **How many rows are in the dataset?**\
There are 5110 cases in this dataset.

8. **What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset**?\
One challenge could be dealing with data imbalance, which might necessitate the identification and application of appropriate techniques and hyperparameters to effectively model the imbalanced dataset. Additionally, determining the optimal approaches for data standardization, handling missing values through imputation, and selecting appropriate modeling strategies will be crucial for achieving the highest level of prediction accuracy.